### PR TITLE
[Inductor cutlass backend] Add bmm support

### PR DESCRIPTION
**Pull request by kadeng**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #121492
* #124577
* #124576
* #124575
* #124574
* #124107
* __->__ #121734

Add support for bmm ( batch matrix multiply ) op through the Cutlass backend.

Test Plan:
 * CI
 * Added test in test_cutlass_backend.py

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by kadeng** [comment]
This PR requires some of the changes which are still upstream in the PR stack to make all tests pass without CUDA IMA errors. Will keep it in draft mode until that's resolved.

File: torch/_inductor/kernel/bmm.py [hunk 1]
```diff
@@ -99,6 +111,23 @@ def tuned_bmm(mat1, mat2, *, layout=None):
...
+            size=WrapperCodeGen.statically_known_list_of_ints_or_none(layout.size),
+        )
+        CUTLASSGemmTemplate.add_cutlass_gemm_choices(choices, out_layout, [mat1, mat2])
+    use_aten = use_aten_gemm_kernels()
+    if len(choices) == 0 and not use_aten:
+        log.warning("No choices for GEMM, using ATen backend as fallback")
+        use_aten = True
+
+    if use_aten:
+        choices.extend([aten_bmm.bind((mat1, mat2), layout)])
```
Comment (Skylion007) [inline torch/_inductor/kernel/bmm.py:130]:
```suggestion
        choices.extend((aten_bmm.bind((mat1, mat2), layout)))
```

Comment (kadeng) [reply torch/_inductor/kernel/bmm.py]: Yep, this is also a bit overkill anyway as it's only a single value to append. I replaced it with a call to append

**Review by eellison — APPROVED** [review]

**Pull request closed without merging**